### DISCUSSION TITLE
Moosefs - files in trash and damaged disks impact

### DISCUSSION THREAD

**Discussion by vishme** [moosefs/moosefs#583] [Q&A]
![IMG-20240907-WA0030](https://github.com/user-attachments/assets/e81d2db6-90f4-4355-a93a-4fd3d57d2706)
It is showing move in ./TRASH

Please help me 🙏

**Comment by jkiebzak** [comment]
Some process, perhaps running in your docker image, deleted those files.

**Comment by xandrus** [comment]
Hi!

If you see missing files with TRASH at the beginning, it means that these files are in TRASH and mostly some chunks are missing.

Are you sure all chunk servers and all disks are connected to the cluster?
Servers TAB and Disks TAB will tell you what is going on.
Also check that there are no hardware failures on the chunk servers. 

By the way - what is the definition of the storage class used for these files?

**Comment by vishme** [comment]
Yes I have 3 Chunk all are working fine ..
I setup moosefs free version in Ubuntu 22.04  last 15 days it have issue
like this.. it is moving data in TRASH and I suprise I search TRASH folder
but no luck.
So is moosefs free version have limitations ?
Is there way to recall all data from trash and or how to recall chunks

**Comment by xandrus** [comment]
Would you be so kind to share the results of this command with us?
`mfscli -H [MFS MASTER IP] -SMF -SCS -SHD -SRS`

To recover files from trash you have to mount meta resource and recover files from there `mfsmount -H [MFS MASER IP] -m /mnt/meta`. Please check chapter 8.13 - "When I delete files or directories, the MooseFS free space size doesn’t change. Why?" in MooseFS manual:
https://moosefs.com/Content/Downloads/moosefs-3-0-users-manual.pdf

But when you see missing files, it means that there are no chunks or the chunks are corrupted or are in the wrong version - so something unusual has happened on your cluster!

**Comment by vishme** [comment]
[mfscli output.txt](https://github.com/user-attachments/files/16986579/mfscli.output.txt)

out put of 
mfscli -H 10.42.0.10 -SMF -SCS -SHD -SRS

In /mnt/mfs/ 2 directory 1 is images which all are showing and that formate is zip another directory is network volumes it have non zip/rar data that all data is disappred. while our tool create new data directory. or may b some techinical glitch.

**Reply by vishme** [reply]
Thanks in advance for support

**Comment by xandrus** [comment]
It looks like you have some errors on two disks.
Are you sure that there are no other hardware issues on these servers or maybe not all chunk servers are connected?
MooseFS do not remove or destroy anything by it own. Some one had to delete these files. And if there are no chunks on the cluster it means that something had happened on the chunk servers. Please definitely verify if your hardware is working correctly.

If some data is very important, I would strongly recommend adding more chunk servers and changing the storage class to 3 for these files.

**Reply by vishme** [reply]
Thank you for your information
My dashboard shown 
missing chunk 478770
and stable chank is 2933779
undergoal 533064

so I m surprise that why stable chunk data not shown in mount drive.





![Screenshot from 2024-09-13 22-13-30](https://github.com/user-attachments/assets/f173fb6f-bf1c-4cc5-a9aa-a9e108cccd29)

**Comment by vishme** [comment]
Also is there way to change status of pending delete chunk data
